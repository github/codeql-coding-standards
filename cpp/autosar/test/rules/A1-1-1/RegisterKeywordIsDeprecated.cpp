void test() {
  register int a; // NON_COMPLIANT
  int b;          // COMPLIANT
}

int foo(register int a) { // NON_COMPLIANT
  return a;
}