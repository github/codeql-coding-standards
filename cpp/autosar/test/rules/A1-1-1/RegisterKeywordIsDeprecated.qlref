rules/A1-1-1/RegisterKeywordIsDeprecated.ql