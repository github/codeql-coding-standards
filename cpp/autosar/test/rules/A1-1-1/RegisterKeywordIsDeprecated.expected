| RegisterKeywordIsDeprecated.cpp:2:16:2:16 | a | Use of the `register` specifier is deprecated in ISO C++14. |
| RegisterKeywordIsDeprecated.cpp:6:22:6:22 | a | Use of the `register` specifier is deprecated in ISO C++14. |
